### ISSUE TITLE
API POST routes returning 405

### ISSUE THREAD

**Issue by lpagliari** [JohnMcLear/ep_comments#41]
tl;dr
When `ep_etherpad-lite/static` is loaded before `ep_comments_page`, API POST routes return `405: Only the HEAD or GET methods are allowed`. Apparently just adding this hook on `ep.json` fixes the issue:

```
"post": ["ep_etherpad-lite/static"],
```

Long story:

I've noticed that sometimes the POST routes on the API were returning `405: Only the HEAD or GET methods are allowed`. As it was kind of a random behaviour, my guessing was that it might be related to the order things were being loaded (other plugins, Etherpad hooks, etc). So I got the logs from a successful and a failing running of Etherpad:

This makes POST route work:
```
[2015-05-20 10:31:03.245] [DEBUG] console - Installed parts:
ep_script_page_view/page_view
ep_disable_lists/ep_disable_lists
ep_font_color/main
ep_script_elements/main
ep_adminpads/ep_adminpads
ep_etherpad-lite/swagger
ep_etherpad-lite/adminsettings
ep_etherpad-lite/adminplugins
ep_etherpad-lite/admin
ep_etherpad-lite/tests
ep_etherpad-lite/socketio
ep_etherpad-lite/errorhandling
ep_etherpad-lite/importexport
ep_etherpad-lite/apicalls
ep_etherpad-lite/webaccess
ep_comments_page/comments_page
ep_etherpad-lite/padreadonly
ep_etherpad-lite/padurlsanitize
ep_etherpad-lite/specialpages
ep_etherpad-lite/i18n
ep_etherpad-lite/static
ep_etherpad-lite/express
```

This makes POST route return 405:
```
[2015-05-20 10:30:14.873] [DEBUG] console - Installed parts:
ep_script_page_view/page_view
ep_script_elements/main
ep_font_color/main
ep_etherpad-lite/swagger
ep_etherpad-lite/adminsettings
ep_etherpad-lite/adminplugins
ep_etherpad-lite/admin
ep_etherpad-lite/tests
ep_etherpad-lite/socketio
ep_etherpad-lite/errorhandling
ep_etherpad-lite/importexport
ep_etherpad-lite/apicalls
ep_etherpad-lite/padreadonly
ep_etherpad-lite/padurlsanitize
ep_etherpad-lite/specialpages
ep_etherpad-lite/i18n
ep_etherpad-lite/static
ep_etherpad-lite/express
ep_disable_lists/ep_disable_lists
ep_etherpad-lite/webaccess
ep_comments_page/comments_page
ep_adminpads/ep_adminpads
```

So my test was basically to get each of the 6 hooks that were called after ep_comments on the successful running (`ep_etherpad-lite/padreadonly`, `ep_etherpad-lite/padurlsanitize`, `ep_etherpad-lite/specialpages`, `ep_etherpad-lite/i18n`, `ep_etherpad-lite/static`, `ep_etherpad-lite/express`) and force them to be a `post` hook for the plugin.

Apparently the hook that is making the routes fail is `ep_etherpad-lite/static`, so to validate that I've tested ep_comments_page with 2 configurations on `ep.json`:

* forcing `static` to be loaded before, and all other candidate hooks after:

  ```
      "pre": ["ep_etherpad-lite/webaccess", "ep_etherpad-lite/static"],
      "post": ["ep_etherpad-lite/padreadonly", "ep_etherpad-lite/padurlsanitize", "ep_etherpad-lite/specialpages", "ep_etherpad-lite/i18n", "ep_etherpad-lite/express"],
  ```

* forcing `static` to be the only hook loaded after:

  ```
      "pre": ["ep_etherpad-lite/webaccess", "ep_etherpad-lite/padreadonly", "ep_etherpad-lite/padurlsanitize", "ep_etherpad-lite/specialpages", "ep_etherpad-lite/i18n", "ep_etherpad-lite/express"],
      "post": ["ep_etherpad-lite/static"],
  ```

And, as expected, the first configuration returned consistently 405, and the second one worked fine.

So it seems that just adding this to `ep.json` would fix the issue:

```
"post": ["ep_etherpad-lite/static"],
```

Note: this is the `curl` command I was using:

```
curl -X POST http://localhost:9001/p/LFP/comments -d "apikey=yourApiKey" -d "name=Author" -d "text=Comment"
```

I'm not exactly sure why that is happening, I could just figure out a way to fix it. Any other ideas?

I'll submit a PR with the fix above if you agree with the solution.

**Comment by JohnMcLear** [comment]
Yea I have no idea on this..   Seems a bit bonkers though...  Might be worth contributing this fix to https://github.com/johnmclear/ep_post_data too ?

**Comment by lpagliari** [comment]
Done.
